egg/core/trainers.py

```diff
@@ -231,6 +237,8 @@ def train_epoch(self):
                 interaction = Interaction.gather_distributed_interactions(interaction)
             interaction = interaction.to("cpu")
             interactions.append(interaction)
+            for callback in self.callbacks:
+                callback.on_batch(interaction, optimized_loss.detach(), data_len)
```

Comment:
don't think detach() is needed here, is it?

Reply:
It depends on the user I guess, If you want to perform some operations on the loss in numpy you need it to be detached. But you can probably do it in the callback

Reply:
yes I would leave it to the user since we're not saving much memory anyway. and we shouldn't assume any training operation is done on the loss in the callback since it won't be backprop through anyway